util/src/main/java/tech/pegasys/teku/util/config/KeyStoreFilesLocator.java

```diff
@@ -0,0 +1,145 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class KeyStoreFilesLocator {
+  private static final Logger LOG = LogManager.getLogger();
+  private final Map<Path, Path> pathMap = new HashMap<>();
+  private final List<String> colonSeparatedPairs;
+  private final String pathSeparator;
+
+  public KeyStoreFilesLocator(final List<String> colonSeparatedPairs, final String pathSeparator) {
+    this.colonSeparatedPairs = colonSeparatedPairs;
+    this.pathSeparator = pathSeparator;
+  }
+
+  public void parse() {
+    for (final String currentEntry : colonSeparatedPairs) {
+      if (!currentEntry.contains(pathSeparator)) {
+        throw new InvalidConfigurationException(
+            "validatorKeys entry ("
+                + currentEntry
+                + ") did not contain key and password separated by '"
+                + pathSeparator
+                + "' as expected.");
+      }
+
+      final List<String> entry = Splitter.on(pathSeparator).limit(2).splitToList(currentEntry);
+      parseEntry(entry.get(0), entry.get(1));
+    }
+  }
+
+  public void parseKeyAndPasswordList(
+      final List<String> keystoreFiles, final List<String> keystorePasswordFiles) {
+    for (int i = 0; i < keystoreFiles.size(); i++) {
+      parseEntry(keystoreFiles.get(i), keystorePasswordFiles.get(i));
+    }
+  }
+
+  void parseEntry(final String keyFileName, final String passwordFileName) {
+    final File keyFile = new File(keyFileName);
+    final File passwordFile = new File(passwordFileName);
+
+    if (!keyFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format("Invalid configuration. Could not find the key file (%s).", keyFileName));
+    }
+    if (!passwordFile.exists()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. Could not find the password file (%s).", passwordFileName));
+    }
+    if (keyFile.isDirectory() != passwordFile.isDirectory()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Invalid configuration. --validator-keys entry (%s"
+                  + pathSeparator
+                  + "%s) must be both directories or both files",
+              keyFileName,
+              passwordFileName));
+    }
+    if (keyFile.isFile()) {
+      pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
+    } else {
+      parseDirectory(keyFile, passwordFile);
+    }
+  }
+
+  void parseDirectory(final File keyDirectory, final File passwordDirectory) {
+    try (Stream<Path> walk = Files.walk(keyDirectory.toPath())) {
+      walk.filter(Files::isRegularFile)
+          .filter(
+              (path) ->
+                  !path.getFileName().toString().startsWith(".")
```

Comment:
I've only just learnt this but we should use `File.isHidden` for this so:
```suggestion
                  !path.toFile().isHidden()
```